protocol/common.go

```diff
@@ -0,0 +1,18 @@
+package protocol
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+)
+
+// ComputeDirectoryIdentity returns the hash of
+// the directory's initial STR as a string.
+// It panics if the STR isn't an initial STR (i.e. str.Epoch != 0).
+func ComputeDirectoryIdentity(str *DirSTR) string {
+	if str.Epoch != 0 {
+		panic(fmt.Sprintf("[coniks] Expect epoch 0, got %x", str.Epoch))
+	}
+	return hex.EncodeToString(crypto.Digest(str.Signature))
```

Comment:
Using a string simplifies indexing for the `AuditLog`. Before this commit, the index needed was represented as a 32-byte array, which worked fine but ended up being a bit messy.